src/include/buffer.h

```diff
@@ -34,7 +34,7 @@
 #include <stdint.h>
 #include <string.h>
 
-#ifndef __CYGWIN__
+#if !defined(__CYGWIN__) && !defined(_WIN32)
```

Comment:
could you prefix the title of your commit message with the subcomponent your are changing ? see https://github.com/ceph/ceph/blob/master/SubmittingPatches.rst#3-describe-your-changes. we don't have a subcomponent dedicated to win32 at this moment. in this case, it could be "include/buffer.h". please see `git log src/include/buffer.h` for more examples.

the same applies to other commits in this PR.

Reply:
Sure. FWIW, this was meant to be part of the next PR but I moved it by mistake.

I'd keep the win32 tag along with the subsystem names, it allows us to keep track of those patches much more easily.

Reply:
There are about 60 commits in this chain, (some of the ones from  the windows.10 branch will be squashed though).

I'll fix their title as we move on and address other comments against the next PRs, I hope that's ok with you.

Reply:
@petrutlucian94 could you address this comment. as the only commit in this PR still have "win32" in the prefix of its title. and i'd suggest remove it for the reason i explained just now.

Reply:
Would you be ok with labeling it in a different way so that we make the distinction between the "win32" tag and subsystems, such as [win32]? That's they way it showed up in the first PR, the one that got merged: https://github.com/ceph/ceph/pull/31981/commits

The reason why I insist on keeping the win32 tag is that it makes it easier to track those commits, which is crucial when having 60 of them. It also provides a scope so that anyone can tell right away that it's about Windows support.

Reply:
@petrutlucian94 why do you want to tag a commit by adding win32 in the **prefix** of its title? what's the use case of this? can those who interested in finding the win32 related commits search the PR by "win32" label? or could you rename the title to something like 

> common: Add dlfcn_compat.h for win32

or anyone interested in this could just search win32 **in** the title?

Reply:
furthermore, i am working on python3 migration, shall we add "python3" in the prefix of the commit title for SEO ? i doubt it. again, the title is not a collection of tags, it just summarizes the change.

Reply:
That's one of the reasons. If it's clearly underlined that it's a tag, e.g. [win32], people are more likely to use it. Otherwise we might get "Add dlfcn_compat.h for Windows" or other similar forms, instead of the same tag.

It also keeps the commit messages shorted and avoids confusion. For example, I have a commit saying "[win32] Initialize WSA and TLS". Without the tag, most probably people wouldn't be able to tell right away what's it about, it's quite unlikely that someone would think that WSA stands for Windows Socket API.

Reply:
then why not be more specific by putting "init WSA and TLS for win32" ?

Reply:
> furthermore, i am working on python3 migration, shall we add "python3" in the prefix of the commit title for SEO ? i doubt it. again, the title is not a collection of tags, it just summarizes the change.

IMO, if it helps, can't see any reason why not. It's not just SEO, it's easier for people going through the commit title, without checking the description.

Reply:
> for win32

I'll have 60 commits ending in with "for win32", that prefix tag seemed cleaner but I can go either way.

Reply:
Lucian, i still think we should not do this in a free-style way. but i agree with you that the title should help people walking through the commits. another thing is that we group the pull requests by its component in our release notes, i don't think our script is smart enough to group the PRs by tags. probably one day, we will have more flexible and structured way to define a title, but i hope we can stick to the format of "<component>: <summary>" before that.

Reply:
> another thing is that we group the pull requests by its component in our release notes

Makes sense, I'll change the title. I'm glad that we had this discussion though so that we don't have to get back to this topic.